src/Tuura/Boolean.hs

```diff
@@ -0,0 +1,86 @@
+module Tuura.Boolean (
+  module Tuura.Boolean.Parser,
+  CNF, DNF, Literal (..),
+  convertToCNF, genConcepts,
+  simplifyCNF, simplifyDNF, convertCNFtoDNF) where
+
+import Tuura.Boolean.Parser
+import Data.List
+import Data.List.Extra
+import Data.Foldable
+import Data.Maybe
+import Data.Ord
+
+type CNF a = [[Literal a]]
+
+type DNF a = [[Literal a]]
+
+data Literal a = Literal { variable :: a, polarity :: Bool } deriving (Eq, Ord)
+
+convertToCNF :: Eq a => (Expr a) -> CNF a
+convertToCNF expr = cnf
+  where
+    vars = toList expr
+    values = mapM (const [False, True]) vars
+    fs = filter (\v -> not $ eval expr (\x -> getValue x vars v)) values
```

Comment:
Also, flip its arguments, so you can write simply `not $ eval expr (getValue vars v)` here.